### PR TITLE
Search: fix zoekt result limits

### PR DESCRIPTION
**Pull request by jtibshirani**
For indexed search, frontend translates a user's query into zoekt options. This
PR makes these fixes to the zoekt result limits, which could help search
performance and memory usage a bit:
* Removes `resultCountFactor`, and instead uses simple default limits for
ShardMaxDocCount and TotalMaxDocCount.
* Updates old logic that increases ShardMaxDocCount and TotalMaxDocCount too
much. The check used the wrong default value, so it always fired and multiplied
the intended value by 50.
* Removes MaxDocDisplayCount adjustment. Before we always added 2000, but this
is not needed because we no longer use these extra results.

Follow-up to #47058

## Test plan

Adapted tests, also played around with large searches to test limit behavior.

### CODE REVIEW

File: internal/search/zoekt/zoekt.go [hunk 1]
```diff
@@ -141,9 +126,19 @@ func (o *Options) ToSearch(ctx context.Context) *zoekt.SearchOptions {
 		k := o.resultCountFactor()
 		searchOpts.ShardMaxMatchCount = 100 * k
 		searchOpts.TotalMaxMatchCount = 100 * k
-		// Ask for 2000 more results so we have results to populate
-		// RepoStatusLimitHit.
-		searchOpts.MaxDocDisplayCount = int(o.FileMatchLimit) + 2000
```
Comment (jtibshirani) [inline internal/search/zoekt/zoekt.go:146]: I poked around and couldn't find where we use these extra results anywhere. To the best of my knowledge, we don't set RepoStatusLimitHit when doing an indexed search.

**Comment by jtibshirani** [comment]
This isn't super high priority, it was just broken and I wanted to unify the old vs. new ranking pathways. That way we can iron out any issues early before enabling advanced ranking.

File: internal/search/zoekt/zoekt.go (hunk 1 above)
Comment (keegancsmith) [reply internal/search/zoekt/zoekt.go:146]: yeah I think this is remnants of the past of when we would give messages about what we found but didn't show to the user.

**Review by keegancsmith — APPROVED** [review]
LGTM! I see you are keeping around the magic k param. What do you think about just using the new ranking path for both old and new rank code paths?

**Comment by jtibshirani** [comment]
@keegancsmith I was unsure about removing `resultCountFactor` because it seemed carefully thought out, and we've had regressions around changing it before: #22779.

But it does seem fine to remove ... the new defaults of 10,000 of 100,000 are an upper bound on what we'd choose using `resultCountFactor`. I'll remove it and check how things look on dot com.

**Pull request merged**